### DISCUSSION TITLE
How I can install Posthog-js in superset?

### DISCUSSION THREAD

**Discussion by vikashrajgupta** [apache/superset#28930] [Q&A / Help]
[Postghog-Js](https://posthog.com/docs/session-replay/iframes#recording-cross-origin-iframes) is an analytics tool and I wanted to install this in an Apache superset.

I have explored the superset project structure and found one of the HTML files i.e. `tail_js_custom_extra.html` where they have mentioned that you can add custom frontend code like Google Analytics etc...

Posthog js also does the interface recording that's why I wanted to use this.

Right now, I have a website let's say `vikash.google.com` in which I have loaded the superset dashboard via Iframe (Embedded SDK) and I wanted to configure Posthog to track the user activity in the recording.

I tried to install the same and build the project but didn't get any kind of error or console message.

That would be great if you could help me get a solution for this.

Here are the steps that I followed:

1. Added console and posthog snippet inside script tag in `tail_js_custom_extra.html`
2. cd superset-frontend
3. npm install
4. npm run build
5. docker down
6. docker up

And then I opened superset.google.com let's say for example, I'm not getting any error or console value that I have added.

_PS: I have already installed Posthog in vikash.google.com and working fine as expected but not able to see the dashboard that is loaded in Iframe._

**Comment by vikashrajgupta** [comment]
@rusackas - In my opinion, you may help me here to solve this. can you please look into this? as this is a blocker at my side.
Thanks in advance!!!

**Reply by rusackas** [reply]
I don't know anything about Posthog really, but it sounds like you're running your host app on (something like) vikash.google.com and Superset on (something like) `superset.google.com`. It seems like that _should_ work, but you need to "add the recordCrossOriginIframes configuration option, and set the correct content security policy"

Or is the embedded Superset just not working at all? If so, was the embedding working before you added Posthog?

It does sound like you're adding the JS in the right part of the code, for whatever that's worth.

**Reply by vikashrajgupta** [reply]
@rusackas - I have added the same `recordCrossOriginIframes` but on the superset side, custom CSS and js code are not reflecting.
Please have a look at the link below for more information about this issue.
https://github.com/apache/superset/discussions/29097

**Reply by vikashrajgupta** [reply]
@rusackas - Can you please help me in this?
